### ISSUE TITLE
Right-click doesn't work in Find > Find what: & Replace > Find what:

### ISSUE THREAD

**Issue by Nthoney** [SubtitleEdit/subtitleedit#8466]
Since version 4.0.5 it is not possible to right-click in the Find > Find what: box.
Also it doesn't work in the Replace > Find what: box.
It does work in the Replace with: box.

Before version 4.0.5 this was never an issue and I'd wished that it would been solved in the next update.
Unfortunately the problem still exist in version 4.0.6.
That's why I'm reporting it now.

Can U please fix this issue?
Thank U very much in advance.

**Comment by niksedk** [comment]
Hi @Nthoney, thx for your input - I've tried to improve latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.6/SubtitleEditBeta.zip
Better?

**Comment by Nthoney** [comment]
Hi Nikolaj,

Thank U for the update.
In the beta version the right-click works fine as before.

I have installed v4.0.6 on my PC and when I copy my Settings.xml to the beta version then the issue reappears.
And when I copy the Settings.xml from the beta version to %AppData%\Subtitle Edit then the right-click works in the installed version 4.0.6.

Can U tell me which line causes the issue?
I've compared the two Settings.xml files a little bit and did searches on "Find", "Replace" and "RightClick", but I couldn't find anything that could be the cause.
And I'm not a nerd (just a bit geeky 😊), so I have no idea what else I could try.

Thank U very much in advance.

Warm regards,
Anthoney

--
Sent with Tuta; enjoy secure & ad-free emails:
https://tuta.com


Jun 4, 2024, 17:54 by ***@***.***:

**Comment by niksedk** [comment]
Thx for testing :)

Sorry, I cannot guess what causes your settings.xml issue....

**Comment by Nthoney** [comment]
Hi Nikolaj,

From Ur answer "Sorry, I cannot guess what causes your settings.xml issue", I concluded that U think that the issue is a local one.
So I did some more testing.

I downloaded a new portable 4.0.6 version and imported step by step all the following adjustments:
- Profiles
- Multiple replace groups
- Advanced Sub Station Alpha styles.
After each import I tested and the issue didn't occur.

Then I compared the two Settings.xml files and the local Settings.xml file had 12 lines more.
I searched where those were and found that the discrepancy was caused by the <FindHistory>


When I removed those 12 lines, the local installation worked fine again.

Then I did a search on the music symbol ♪ (Alt 1-3) and it broke the Search box.
Both in the local installation and in the portable version, when I did a search on "♪" it finds the first instance.
But when I click on Find next the second time, the music symbol disappears.
This doesn't happen when I press Enter on my keyboard.
Then it find the next instance.
But in all cases the right-click option becomes unavailable.
When I open the Find menu again, then the music symbol is visible and Find next works properly.
But still, right-click doesn't work.

I've also tested it in Ur beta version and the behavior is also faulty.

Then I tested it in Subtitle Edit versions 3.6.13 through 4.0.5.
In all versions the music symbol disappears after the first search.
The inability to right-click started with version 4.0.5.
So my conclusion is that this is not a local issue, but a structural one.

Wish this helps 😉

Warm regards,
Anthoney

-- 
 Sent with Tuta; enjoy secure & ad-free emails: 
 https://tuta.com


Jun 5, 2024, 06:42 by ***@***.***:
